frontend/src/views/Announcements/AnnouncementsList.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  Button,
+  Icon,
+  IconButton,
+  Stack,
+  Toolbar,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+import {
+  AnnouncementCreateRequestDto,
+  AnnouncementCreateResponseDto,
+  AnnouncementDeleteResponseDto,
+  AnnouncementFindAllResponseDto,
+  AnnouncementUpdateRequestDto,
+  AnnouncementUpdateResponseDto,
+  DtoAnnouncement,
+} from '@osk/shared';
+import { Box, Flex } from 'reflexbox';
+import useSWR from 'swr';
+import { UserRole } from '@osk/shared/src/types/user.types';
+import { useState } from 'react';
+import { FullPageLoading } from '../../components/FullPageLoading/FullPageLoading';
+import { GeneralAPIError } from '../../components/GeneralAPIError/GeneralAPIError';
+import { useActionModal } from '../../hooks/useActionModal/useActionModal';
+import { AnnouncementCard } from './AnnouncementCard/AnnouncementCard';
+import { useAuth } from '../../hooks/useAuth/useAuth';
+import { ActionModal } from '../../components/ActionModal/ActionModal';
+import { useCommonSnackbars } from '../../hooks/useCommonSnackbars/useCommonSnackbars';
+import { useMakeRequestWithAuth } from '../../hooks/useMakeRequestWithAuth/useMakeRequestWithAuth';
+import { AnnouncementFormModal } from './AnnouncementFormModal/AnnouncementFormModal';
+import { AnnouncementFormSchema } from './AnnouncementFormModal/AnnouncementFormModal.schema';
+
+export const AnnouncementsList = () => {
```

Comment:
General comment to the whole component — why don't we break the business logic in this component into multiple hooks? Namely `useEditAnnouncement`, `useCreateAnnouncement`, and `useDeleteAnnouncement`.

Reply:
Cause I am nowhere near having react developer intuition. Will work on this suggestion though - update coming